consensus/src/simplex/actors/voter/actor.rs

```diff
@@ -341,13 +341,26 @@ impl<E: Clock, S: Scheme, D: Digest> Round<E, S, D> {
         Some(finalization)
     }
 
-    /// Returns whether at least one honest participant has notarized a proposal.
-    pub fn at_least_one_honest(&self) -> Option<View> {
-        if self.notarizes.len() <= self.scheme.participants().max_faults() as usize {
-            return None;
+    /// Returns true if there is evidence that at least one honest participant
+    /// notarized the proposal for this view. This is true if:
+    /// - a finalization for this view exists (implies an honest notarization existed), or
+    /// - a notarization certificate for this view exists, or
+    /// - the number of notarize votes exceeds the maximum number of faulty participants (f),
+    ///   which implies at least one vote came from an honest participant.
+    ///
+    /// Used to decide whether we can safely backfill missing certificates implied by a proposal.
+    pub fn has_honest_notarization(&self) -> bool {
```

Comment:
nit: `proposal_has_valid_ancestry`

I think `has_honest_notarization` isn't quite specific enough. It sounds like [1] it returns whether or not a round has a notarization, [2] implies that there can be dishonest notarizations, and [3] doesn't indicate there can be both notarizations/nullifications to consider as part of the ancestry.